torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -1115,14 +1115,20 @@ def block_weight(cls, W, new_size, padding):
 
     @classmethod
     def pack_vnni_weight(cls, W, micro_gemm, new_size):
+        # WOQ INT4 weights are reordered in microkernel so do not pack them here
+        should_pack = (
+            micro_gemm.get_b_layout() != LayoutType.NORMAL
+            and not micro_gemm.is_woq_int4()
```

Comment:
If the packing happens at the runtime (in the micro-kernel), should we set the layout of "b" to be NORMAL instead?

Reply:
Thanks for the comments. Layout of b is not normal. It's blocked but not VNNI. At runtime, we load two lines of K dim and shuffle them to VNNI layout for computation.